3p/ampcontext.js

```diff
@@ -29,6 +29,9 @@ export class AbstractAmpContext {
     dev().assert(!this.isAbstractImplementation_(),
         'Should not construct AbstractAmpContext instances directly');
 
+    /** @visibleForTesting */
+    this._ampInternalIsNewImpl = true;
```

Comment:
Yeah, the experiment flag works only when `opt_transient` set to false. I realized this by setting this property.

In the default implementation the property is not set so it's falsey.